src/compiler/watchPublic.ts

```diff
@@ -264,6 +264,7 @@ export interface ProgramHost<T extends BuilderProgram> {
  *
  * @internal
  */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 export interface ProgramHost<T extends BuilderProgram> {

```

Comment:
Can you bring in the change from https://github.com/microsoft/TypeScript/pull/55139/files#diff-ca579690d4b8ab4c83f5d18e3e8797d11a4c814ce32afe840eb3d7b797c46df5 instead? It's more typical for us to be using internal properties than declaration merging for this purpose.